### PR TITLE
Prebid Core: Library support

### PR DESCRIPTION
**Pull request by karimMourra**
## Type of change
- [x] Build related changes

## Description of change
 This PR allows modules to share code outside of core by introducing `Libraries`. With these changes, code outside of `core` can be shared without duplication in the build. 

As an example, the `extraUtils` library is introduced. The `getOrigin` function from the core utils has been moved to this library, and the library is now listed as a dependency of the 3 modules that use the util. 

Note that when the dist is built or bundled, the `getOrigin` function is not duplicated.
You can test with:
`gulp build --modules=ooloAnalyticsAdapter,resetdigitalBidAdapter`
`gulp bundle --modules=ooloAnalyticsAdapter,resetdigitalBidAdapter`

## Other information
Helps with issue #7936
Needed for #7264
@dgirardi for review

### CODE REVIEW

**Comment by muuki88** [comment]
This is amazing work. Thanks a lot 🎉

**Review by dgirardi — COMMENTED** [review]
Thanks for putting this together! I disagree only with the name choice for the first library.
Also noting that this does not completely address https://github.com/prebid/Prebid.js/issues/7936 (which is perfectly fine - I just wouldn't want that to be closed prematurely).

File: libraries/extraUtils/index.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+/**
+ * Returns the origin
+ */
+export function getOrigin() {
```
[the remark this answers is not in the document]
Comment (karimMourra) [reply libraries/extraUtils/index.js]: addressed

File: modules/.submodules.json [hunk 2]
```diff
@@ -1,75 +1,87 @@
...
+      "enrichmentFpdModule",
+      "validationFpdModule"
+    ]
+  },
+  "libraries": {
+    "extraUtils": {
+      "files": [
+        "./index.js"
+      ],
+      "dependants": [
```
[the remark this answers is not in the document]
Comment (karimMourra) [reply modules/.submodules.json:80]: makes sense, I'm happy to explore that in the near future when cycles free up

[the remark this answers is not in the document]
Comment (mkendall07) [reply modules/.submodules.json:78]: 👍 I could see this dependents list getting out of hand quickly.

File: libraries/getOrigin/index.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
+/**
+ * Returns the origin
+ */
+export function getOrigin() {
```
Comment (mkendall07) [inline libraries/getOrigin/index.js:4]: is there any restrictions on library files? What happens if you want to import another file from the library file?

**Review by mkendall07 — APPROVED** [review]
Overall looks great. Just left one question. Good work solving a long standing problem with the build system!

File: libraries/getOrigin/index.js (hunk 3 above)
Comment (patmmccann) [reply libraries/getOrigin/index.js:4]: We'll probably need to cover this at a later date, with some best practices or boundaries

**Pull request merged**